runtime-sdk/modules/evm/src/precompile/subcall.rs

```diff
@@ -0,0 +1,522 @@
+use ethabi::{ParamType, Token};
+use evm::{
+    executor::stack::{PrecompileFailure, PrecompileHandle, PrecompileOutput},
+    ExitError, ExitSucceed,
+};
+
+use crate::backend::EVMBackendExt;
+use oasis_runtime_sdk::{
+    module::CallResult, modules::core::Error, subcall, types::transaction::CallerAddress,
+};
+
+use super::{record_linear_cost, PrecompileResult};
+
+/// A subcall validator which prevents any subcalls from re-entering the EVM module.
+struct ForbidReentrancy;
+
+impl subcall::Validator for ForbidReentrancy {
+    fn validate(&self, info: &subcall::SubcallInfo) -> Result<(), Error> {
+        if info.method.starts_with("evm.") {
+            return Err(Error::Forbidden);
+        }
+        Ok(())
+    }
+}
+
+// XXX: fix these costs
+const SUBCALL_BASE_COST: u64 = 1_000;
```

Comment:
These costs are in addition to the costs itself by the subcall right? So likely can be quite small? Basically mostly for the parsing of arguments, right?

Reply:
Yeah, exactly, it is just for argument CBOR parsing.